src/lib.rs

```diff
@@ -150,3 +159,88 @@ fn impl_glium_vertex_derive(ast: &DeriveInput) -> TokenStream {
 
     stream.into()
 }
+
+#[proc_macro_derive(Uniforms, attributes(glium))]
+pub fn glium_uniforms_derive(input: TokenStream) -> TokenStream {
+    let ast = syn::parse(input).unwrap();
+    impl_glium_uniforms_derive(&ast)
+}
+
+fn impl_glium_uniforms_derive(ast: &DeriveInput) -> TokenStream {
+    let struct_name = &ast.ident;
+    let generics = &ast.generics;
+    let (impl_generics, ty_generics, where_clause) = generics.split_for_impl();
+
+    let fields = match ast.data {
+        Data::Struct(DataStruct {
+            fields: Fields::Named(ref fields),
+            ..
+        }) => {
+            &fields.named
+        },
+        _ => {
+            panic!("#[derive(Uniforms)] only defined for structs.");
+        }
+    };
+
+    let bindings = fields.iter().map(|field| {
+        let attrs = field.attrs.iter()
+            .flat_map(Attribute::parse_meta)
+            .flat_map(|meta| {
+                match meta {
+                    Meta::List(MetaList { ref path, ref nested, .. }) => {
+                        let ident = path.get_ident().expect("Expected ident got path");
+
+                        if ident == "glium" {
+                            nested.iter().cloned().collect()
+                        } else {
+                            Vec::new()
+                        }
+                    },
+                    _ => {
+                        Vec::new()
```

Comment:
Maybe you could put this Vec::new() at the end of the closure and then drop both the else and compact this arm to just {}? The collect call would get a return at the front.